web/src/message_fetch.js

```diff
@@ -52,6 +52,7 @@ const consts = {
 
     // Parameters for asking for more history in the recent view.
     recent_view_fetch_more_batch_size: 2000,
+    recent_view_minimum_load_more_fetch_size: 50000,
```

Comment:
I tested this commit locally with this hack, and it mostly worked, but one thing I noticed is that the date that we've fetched until didn't seem to update as it did each batch of a few messages -- it just showed the same date. Not sure if it's a development environment data confusion quirk, but seemed like a real bug. Can you investigate?

```
diff --git a/web/src/message_fetch.js b/web/src/message_fetch.js
index 64367320a4..193585d696 100644
--- a/web/src/message_fetch.js
+++ b/web/src/message_fetch.js
@@ -37,22 +37,22 @@ const consts = {
     narrowed_view_forward_batch_size: 100,
 
     // Initial backfill parameters to populate message history.
-    initial_backfill_fetch_size: 1000,
-    catch_up_batch_size: 2000,
+    initial_backfill_fetch_size: 1,
+    catch_up_batch_size: 2,
     // We fetch at least minimum_initial_backfill_size messages of
     // history, but after that will stop fetching at either
     // maximum_initial_backfill_size messages or
     // target_days_of_history days, whichever comes first.
-    minimum_initial_backfill_size: 9000,
-    maximum_initial_backfill_size: 25000,
+    minimum_initial_backfill_size: 0,
+    maximum_initial_backfill_size: 0,
     target_days_of_history: 180,
     // Delay in milliseconds after processing a catch-up request
     // before sending the next one.
     catch_up_backfill_delay: 150,
 
     // Parameters for asking for more history in the recent view.
-    recent_view_fetch_more_batch_size: 2000,
-    recent_view_minimum_load_more_fetch_size: 50000,
+    recent_view_fetch_more_batch_size: 2,
+    recent_view_minimum_load_more_fetch_size: 200,
 };
 
```

Reply:
Looks like `set_oldest_message_date` is the function that changes that date; so likely shouldn't be too hard to print-debug.

Reply:
Oh, it's this logic blocking the update:

```
                                                                                                        
    // Until we've found the newest message, we only show the banner with a messages                    
    // explaining we're still fetching messages. We don't allow the user to fetch                       
    // more messages.                                                                                   
    if (loading_state === SOME_MESSAGES_LOADED || is_backfill_in_progress) {                            
        return;                                                                                         
    }                                                                                                   
                                                                                                        
```

Reply:
Fix should be this:

``` diff
     // Until we've found the newest message, we only show the banner with a messages
     // explaining we're st     // Until we've found the newest message, we only show the banner with a messages
     // explaining we're still fetching messages. We don't allow the user to fetch
     // more messages.
-    if (loading_state === SOME_MESSAGES_LOADED || is_backfill_in_progress) {
+    if (loading_state === SOME_MESSAGES_LOADED) {
         return;
     }
 
-    const $button = $(".recent-view-load-more-container .fetch-messages-button");
-    const $button_label = $(".recent-view-load-more-container .button-label");
     const $banner_text = $(".recent-view-load-more-container .last-fetched-message");
-
-    $button.toggleClass("notvisible", false);
-
     const time_obj = new Date(oldest_message_timestamp * 1000);
     const time_string = timerender.get_localized_date_or_time_for_format(
         time_obj,
@@ -218,6 +213,15 @@ function update_load_more_banner() {
     );
     $banner_text.text($t({defaultMessage: "Showing messages since {time_string}."}, {time_string}));
 
+    if (is_backfill_in_progress) {
+        // Keep the button disabled and the loading indicator running
+        // until we've finished our recursive backfill.
+        return;
+    }
+    const $button = $(".recent-view-load-more-container .fetch-messages-button");
+    const $button_label = $(".recent-view-load-more-container .button-label");
+    $button.toggleClass("notvisible", false);
+
     $button_label.toggleClass("invisible", false);
     $button.prop("disabled", false);
     loading.destroy_indicator(
ill fetching messages. We don't allow the user to fetch
     // more messages.
-    if (loading_state === SOME_MESSAGES_LOADED || is_backfill_in_progress) {
+    if (loading_state === SOME_MESSAGES_LOADED) {
         return;
     }
 
-    const $button = $(".recent-view-load-more-container .fetch-messages-button");
-    const $button_label = $(".recent-view-load-more-container .button-label");
     const $banner_text = $(".recent-view-load-more-container .last-fetched-message");
-
-    $button.toggleClass("notvisible", false);
-
     const time_obj = new Date(oldest_message_timestamp * 1000);
     const time_string = timerender.get_localized_date_or_time_for_format(
         time_obj,
@@ -218,6 +213,15 @@ function update_load_more_banner() {
     );
     $banner_text.text($t({defaultMessage: "Showing messages since {time_string}."}, {time_string}));
 
+    if (is_backfill_in_progress) {
+        // Keep the button disabled and the loading indicator running
+        // until we've finished our recursive backfill.
+        return;
+    }
+    const $button = $(".recent-view-load-more-container .fetch-messages-button");
+    const $button_label = $(".recent-view-load-more-container .button-label");
+    $button.toggleClass("notvisible", false);
+
     $button_label.toggleClass("invisible", false);
     $button.prop("disabled", false);
     loading.destroy_indicator(
```